crates/nu-protocol/src/ty.rs

```diff
@@ -100,3 +117,44 @@ impl Display for Type {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::Type;
+    use strum::IntoEnumIterator;
+
+    mod subtype_relation {
+        use super::*;
+
+        #[test]
+        fn test_reflexivity() {
+            for ty in Type::iter() {
+                assert!(ty.is_subtype(&ty));
+            }
+        }
+
+        #[test]
+        fn test_any_is_top_type() {
+            for ty in Type::iter() {
+                assert!(ty.is_subtype(&Type::Any));
+            }
+        }
+
```

Comment:
I haven't thought about whether `Nothing` should be the Bottom type and I didn't need that to be true.

Reply:
Are you wondering if Nothing is a subtype of Any?

Reply:
I think the question is whether we want to declare that `Nothing` is a subtype of every other type. But there might also be some theoretical questions about whether the `Nothing` type is " inhabitable"?